cms/server/admin/handlers/modelsolution.py

```
@@ -0,0 +1,563 @@
+    """
+    @require_permission(BaseHandler.AUTHENTICATED)
+    def get(self, meta_id, dataset_id=None):
+        meta = self.safe_get_item(ModelSolutionMeta, meta_id)
+
+        if dataset_id is None:
+            dataset_id = meta.dataset_id
+        else:
+            try:
+                dataset_id = int(dataset_id)
+            except ValueError:
+                raise tornado.web.HTTPError(400, "Invalid dataset ID")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)